pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -448,3 +448,12 @@ def reconciliate_processes(self, trace: str):
                 os.kill(pid, signal.SIGKILL)
             shutil.rmtree(sync_dir)
             raise DeadlockDetectedException(f"DeadLock detected from rank: {self.global_rank} \n {trace}")
+
+    def teardown(self) -> None:
+        if self.on_gpu:
+            # GPU teardown
+            self.lightning_module.cpu()
+            # clean up memory
+            torch.cuda.empty_cache()
+            # delete ``ddp wrapper`` as the reducer is holding some cuda memory back.
+            self._model = None
```

Comment:
I think we could do this in the parent directly with 

`self.model = None`

we do have a getter/setter for it

```
    @property
    def model(self) -> Module:
        """Returns the potentially wrapped LightningModule"""
        return self._model

    @model.setter
    def model(self, new_model: Module) -> None:
        self._model = new_model
```

Although now the typing annotation would be wrong as it should include `None`

Should we also just move to cpu instead of setting it to None?

Reply:
Done !